api/tests/integration/domain/services/smart-random/smart-random_test.js

```diff
@@ -661,20 +738,65 @@ describe('Integration | Domain | Stategies | SmartRandom', () => {
           })
         ];
 
-        sinon.stub(_, 'sample').callsFake(function fakeFn(array) {
-          if (array.includes(challengeUrl_2)) {
-            return challengeUrl_2;
-          }
-          return array[0];
-        });
-
         // when
-        const { nextChallenge } = SmartRandom.getNextChallenge({ targetSkills, challenges, knowledgeElements, lastAnswer });
+        const { nextChallenge } = SmartRandom.getNextChallenge({
+          targetSkills,
+          challenges,
+          knowledgeElements,
+          lastAnswer,
+          assessmentId
+        });
 
         // then
         expect(nextChallenge).to.deep.equal(challengeWeb_2_3);
       });
     });
 
+    context('when there is multiply skills and challenge', () => {
```

Comment:
'when there is multiple skills and challenge' ?

Reply:
+1